### PR TITLE
Dev 3872 implement button above contribution list in portal

### PR DESCRIPTION
**Pull request by GuiMend**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)
n/a
#### What's this PR do?
- Adds contact information popover in Portal
#### Why are we doing this? How does it help us?
- Helping donors contact the Org if necessary
#### Are there detailed, specific, step-by-step testing instructions in the associated ticket?
yes
#### Did this PR make changes to the user interface that may require changes to the user-facing docs?
yes
#### Have automated unit tests been added? If not, why?
yes
#### How should this change be communicated to end users?
Contact information now available for donors in the portal
#### Are there any smells or added technical debt to note?
no
#### Has this been documented? If so, where?
no
#### What are the relevant tickets? Add a link to any relevant ones.
https://news-revenue-hub.atlassian.net/browse/DEV-3872
#### Do any changes need to be made before deployment to production (adding environment variables, for example)? If so, open a ticket and link it to the ticket for this PR and list it here:
no
#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:
https://news-revenue-hub.atlassian.net/browse/DEV-3871

### CODE REVIEW

File: spa/src/components/portal/ContributionsList/ContactInfoPopover/ContactInfoPopover.styled.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  && {
+    position: absolute;
+    top: 0;
+    right: 0;
+    height: 28px;
+    width: 28px;
+    padding: 0;
+    display: flex;
+    align-items: center;
+    justify-content: center;
```
Comment (nrh-cklimas) [inline spa/src/components/portal/ContributionsList/ContactInfoPopover/ContactInfoPopover.styled.ts:15]: It doesn't appear that these three lines are needed--if I remove them, the icon is still centered.

File: spa/src/components/portal/ContributionsList/ContactInfoPopover/ContactInfoPopover.styled.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      fill: ${({ theme }) => theme.basePalette.greyscale.white};
+    }
+
+    @media (${(props) => props.theme.breakpoints.tabletLandscapeDown}) {
+      right: 20px;
+    }
+  }
+`;
+
+export const TitleTypography = styled(MuiTypography)`
```
Comment (nrh-cklimas) [inline spa/src/components/portal/ContributionsList/ContactInfoPopover/ContactInfoPopover.styled.ts:27]: I want to get Typography as a base component eventually, but I think until then, let's make this and `ContactTypography` `styled.p` instead.

File: spa/src/components/portal/ContributionsList/ContactInfoPopover/ContactInfoPopover.test.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    tree({ page: { revenue_program: {} } as any });
+    expect(document.body.textContent).toBe('');
+  });
+
+  it.each(['contact_phone', 'contact_email'])('renders contact info popover if %s is defined', (field) => {
+    tree({ page: { revenue_program: { [field]: 'mock' } } as any });
+    expect(screen.getByLabelText('Open contact info')).toBeInTheDocument();
+  });
+
+  it('shows contact info', () => {
```
Comment (nrh-cklimas) [inline spa/src/components/portal/ContributionsList/ContactInfoPopover/ContactInfoPopover.test.tsx:32]: We should test that if phone or email are empty strings, that the `'Phone:'` or `'Email:'` text isn't present. e.g. I'm envisioning a regression where the labels show but have no link.

File: spa/src/components/portal/ContributionsList/ContactInfoPopover/ContactInfoPopover.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+};
+
+const ContactInfoPopoverPropTypes = {
+  page: PropTypes.object
+};
+
+ContactInfoPopover.propTypes = ContactInfoPopoverPropTypes;
+
+export default ContactInfoPopover;
+export type { ContactInfoPopoverProps };
```
Comment (nrh-cklimas) [inline spa/src/components/portal/ContributionsList/ContactInfoPopover/ContactInfoPopover.tsx:78]: Don't think you need this line--change line 17 to be `export interface ContactInfoPopoverProps`.

File: spa/src/components/portal/ContributionsList/ContactInfoPopover/ContactInfoPopover.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  ContactRow,
+  ContactTypography,
+  StyledLink,
+  TitleTypography
+} from './ContactInfoPopover.styled';
+import { ContributionPage } from 'hooks/useContributionPage';
+
+interface ContactInfoPopoverProps extends InferProps<typeof ContactInfoPopoverPropTypes> {
+  page?: ContributionPage;
+}
```
Comment (nrh-cklimas) [inline spa/src/components/portal/ContributionsList/ContactInfoPopover/ContactInfoPopover.tsx:19]: I think it'd be better (looser coupling between components) if the component takes the RP as prop--its job is to display contact info for an RP and doesn't need to know about a contribution page.

File: spa/src/components/portal/ContributionsList/ContactInfoPopover/ContactInfoPopover.tsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    setAnchorEl(event.currentTarget);
+  }, []);
+
+  if (!hasContactInfo) {
+    return null;
+  }
+
+  return (
+    <>
+      <ContactInfoButton color="primaryDark" onClick={showContactInfoPopover} aria-label="Open contact info">
```
Comment (nrh-cklimas) [inline spa/src/components/portal/ContributionsList/ContactInfoPopover/ContactInfoPopover.tsx:36]: Appreciate adding an ARIA label, but it needs to vary based on the state of the popup.

File: spa/src/components/portal/ContributionsList/ContactInfoPopover/ContactInfoPopover.tsx
Comment (nrh-cklimas) [inline spa/src/components/portal/ContributionsList/ContactInfoPopover/ContactInfoPopover.tsx:1]: Could this component be added to Storybook showing the different display possibilities (phone only, email only, both phone and email)?

File: spa/src/components/portal/ContributionsList/ContactInfoPopover/ContactInfoPopover.styled.ts [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    fill: ${({ theme }) => theme.basePalette.greyscale.black};
+    height: 24px;
+    width: 24px;
+  }
+`;
+
+export const StyledLink = styled.a`
+  color: ${({ theme }) => theme.basePalette.primary.engineBlue};
+  font-weight: 500;
+`;
```
Comment (nrh-cklimas) [inline spa/src/components/portal/ContributionsList/ContactInfoPopover/ContactInfoPopover.styled.ts:74]: I talked with Rachel about this--we are soon going to change the hyperlink blue to engine blue, but I want to do this as an app-wide change. In the meantime, please use the base Link component without any custom color.

**Review by nrh-cklimas — CHANGES_REQUESTED** [review]

File: spa/src/components/portal/ContributionsList/ContactInfoPopover/ContactInfoPopover.tsx
Comment (GuiMend) [reply spa/src/components/portal/ContributionsList/ContactInfoPopover/ContactInfoPopover.tsx:1]: Added it to storybook, but to see the popover we need to click the question mark button

**Review by nrh-cklimas — APPROVED** [review]

**Pull request merged**